### ISSUE TITLE
Some fonts causes load error "already has key" with Safari

### ISSUE THREAD

**Issue by casperno** [nodebox/opentype.js#281]
Loading MuseoSans300Italic.otf or BrandonText_Medium.otf causes the
`entriesToObject` method to crash.

## Current Behavior
entriesToObject, in Safari seems to fail if it gets an entries list with numbered keys out of order. Eg. 
```
entries = 
[
[0, [380]],
[1, [647]],
[1200, [648]],
[2, [649]]
]

```
will crash when trying to add key = 2 (last in array). It seems like Safari has an error that when adding o[1200]= [648], it also adds o[2] = NaN. This causes o.hasOwnProperty(key) to trigger an error.

## Possible Solution
Is the `o.hasOwnProperty(key)` test necessary? Can it be expanded to do` o.hasOwnProperty(key) && !isNaN(o[key])`?
Both work for the font I've encountered problems with.

## Your Environment

* Version used: 0.6.9
* Font used: MuseoSans300Italic.otf, BrandonText_Medium.otf
* Browser Name and version: Safari 10.1
* Operating System and version (desktop or mobile): Macos 10.12.4

**Comment by fdb** [comment]
Thanks! Haven't seen that issue yet. 

Can you upload the specific font files, license permitting? I did a quick search for MuseoSans300Italic.otf but the one I found had no problems.

**Comment by casperno** [comment]
I'm not comfortable with uploading the font here, but I could send it to you directly provided you delete it after testing.

**Comment by fdb** [comment]
That would help. My e-mail address is frederik@debleser.be.

**Comment by casperno** [comment]
Font sent

**Comment by neiltron** [comment]
I'm also seeing this issue in Safari using Regina Black. It seems to be happening when calling `opentype.load` twice on the same font (though I've only seen it happen with this one font). @casperno's proposed works for me.

**Comment by kontur** [comment]
Would love to see this fix merged in since I am having a similar issue reported in my Wordpress plugin. 
It is possible this comes from a duplicate font in this instance as well, since it's once defined via a Wordpress theme and then the same font is loaded via a Wordpress plugin, thus resulting in a double definition. I have not been able to properly confirm this though, since it's not my site or font the issue happens on.

**Comment by philippkuehn** [comment]
Any news on this?

**Comment by fdb** [comment]
I was on holiday in the last weeks. I just fixed this, and release 0.7.3, which contains the fix, should appear on npm soon.
